repo-projects/runtime.proj

```diff
@@ -12,37 +15,45 @@
+    <BuildArgs Condition="'$(DotNetBuildSourceOnly)' == 'true' and '$(DotNetBuildUseMonoRuntime)' != 'true'">$(BuildArgs) --bootstrap</BuildArgs>
+
+    <!-- If this build is not a "special flavor" build, build everything that can be built for this target. -->
+    <BuildArgs Condition="'$(PgoInstrument)' != 'true' and '$(DotNetBuildUseMonoRuntime)' != 'true'">$(BuildArgs) /p:DotNetBuildAllRuntimePacks=true</BuildArgs>
 
     <!-- When we're actually doing signing and the ESRP tool is available, forward down the path to the repo builds. -->
     <BuildArgs Condition="'$(Sign)' == 'true' and '$(ForceDryRunSigning)' != 'true' and '$(DotNetEsrpToolPath)' != ''">$(BuildArgs) /p:DotNetEsrpToolPath=$(DotNetEsrpToolPath)</BuildArgs>
 
-    <!-- Needed until https://github.com/dotnet/runtime/issues/109329 is fixed. -->
-    <BuildArgs>$(BuildArgs) /p:NetCoreAppToolCurrentVersion=10.0</BuildArgs>
-
-    <!-- Temporarily disable building tests until the next runtime codeflow with an updated SDK. -->
-    <DotNetBuildTestsOptOut>true</DotNetBuildTestsOptOut>
+    <!-- Pass through special build modes controlled by properties -->
+    <BuildArgs Condition="'$(WasmEnableThreads)' == 'true'">$(BuildArgs) /p:WasmEnableThreads=true</BuildArgs>
+    <BuildArgs Condition="'$(MonoEnableLLVM)' == 'true'">$(BuildArgs) /p:MonoEnableLLVM=true</BuildArgs>
+    <BuildArgs Condition="'$(MonoAOTEnableLLVM)' != ''">$(BuildArgs) /p:MonoAOTEnableLLVM=$(MonoAOTEnableLLVM)</BuildArgs>
+    <BuildArgs Condition="'$(MonoBundleLLVMOptimizer)' != ''">$(BuildArgs) /p:MonoBundleLLVMOptimizer=$(MonoBundleLLVMOptimizer)</BuildArgs>
+    <BuildArgs Condition="'$(PgoInstrument)' == 'true'">$(BuildArgs) $(FlagParameterPrefix)pgoinstrument</BuildArgs>
 
-    <!-- Explicitly disable RID-specific artifacts when it's not enabled. Runtime's build defaults to using RID-specific artifacts, so we must explicitly opt-out. -->
-    <EnableDefaultRidSpecificArtifacts Condition="'$(EnableDefaultRidSpecificArtifacts)' != 'true'">false</EnableDefaultRidSpecificArtifacts>
+    <!-- Handle system libraries -->
+    <UseSystemLibs Condition="'$(UseSystemLibs)' != ''">+$(UseSystemLibs)</UseSystemLibs>
+    <BuildArgs Condition="$(UseSystemLibs.Contains('+brotli'))">$(BuildArgs) --cmakeargs -DCLR_CMAKE_USE_SYSTEM_BROTLI=true</BuildArgs>
+    <BuildArgs Condition="$(UseSystemLibs.Contains('+libunwind'))">$(BuildArgs) --cmakeargs -DCLR_CMAKE_USE_SYSTEM_LIBUNWIND=true</BuildArgs>
+    <!-- TODO: llvm-libunwind -->
+    <!-- TODO: LinuxTracepoints -->
+    <BuildArgs Condition="$(UseSystemLibs.Contains('+rapidjson'))">$(BuildArgs) --cmakeargs -DCLR_CMAKE_USE_SYSTEM_RAPIDJSON=true</BuildArgs>
+    <BuildArgs Condition="$(UseSystemLibs.Contains('+zlib'))">$(BuildArgs) --cmakeargs -DCLR_CMAKE_USE_SYSTEM_ZLIB=true</BuildArgs>
+    <BuildArgs Condition="$(UseSystemLibs.Contains('-lttng'))">$(BuildArgs) /p:FeatureXplatEventSource=false</BuildArgs>
```

Comment:
this should probably be
```suggestion
    <BuildArgs Condition="$(UseSystemLibs.Contains('+lttng'))">$(BuildArgs) /p:FeatureXplatEventSource=false</BuildArgs>
```

Reply:
No idea. Straight copy from https://github.com/dotnet/runtime/blob/dd75c45c123055baacd7aa4418f425f412797a29/eng/DotNetBuild.props#L90

Reply:
This is meant to be '-' (minus) for not using lttng.